lib/collection/property.js

```diff
@@ -319,6 +339,49 @@ _.assign(Property, /** @lends Property */ {
         return _.mergeWith({}, obj, customizer);
     },
 
+    /**
+     * Similar to `replaceSubstitutionsIn` but runs asynchronously
+     * and supports async value functions
+     *
+     * @param {Object} obj -
+     * @param {Array.<VariableList|Object>} variables -
+     * @returns {Object}
+     */
+    replaceSubstitutionsInLazy: async function (obj, variables) {
+        // if there is nothing to replace, we move on
+        if (!(obj && _.isObject(obj))) {
+            return obj;
+        }
+
+        // convert the variables to a substitutor object (will not reconvert if already substitutor)
+        variables = Substitutor.box(variables, Substitutor.DEFAULT_VARS);
+
+        const promises = [];
```

Comment:
@appurva21 Thoughts on supporting async updates this way? This feels weird but I didn't want to touch the `_.mergeWith` and `customizer` fns. Or do you think it's okay if we write a custom implementation for `mergeWith` which supports async replaces ?